src/annotate/celltypist/script.py

```diff
@@ -0,0 +1,108 @@
+import sys
+import logging
+import celltypist
+import mudata as mu
+import re
+import numpy as np
+
+## VIASH START
+par = {
+    "input": "resources_test/pbmc_1k_protein_v3/pbmc_1k_protein_v3_filtered_feature_bc_matrix_log_normalized.h5mu",
+    "output": "output.h5mu",
+    "modality": "rna",
+    "reference": "resources_test/annotation_test_data/TS_Blood_filtered.h5mu",
+    "model": None,
+    "reference_obs_target": "cell_ontology_class",
+    "check_expression": False,
+    "feature_selection": True,
+    "majority_voting": True,
+    "output_compression": "gzip",
+    "var_query_gene_names": None,
+    "var_reference_gene_names": "ensemblid",
+    "input_layer": None,
+    "reference_layer": None,
+    "output_obs_predictions": "celltypist_pred",
+    "output_obs_probabilities": "celltypist_probability",
+}
+meta = {
+}
+## VIASH END
+
+# START TEMPORARY WORKAROUND setup_logger
+def setup_logger():
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    console_handler = logging.StreamHandler(sys.stdout)
+    logFormatter = logging.Formatter("%(asctime)s %(levelname)-8s %(message)s")
+    console_handler.setFormatter(logFormatter)
+    logger.addHandler(console_handler)
+
+    return logger
+# END TEMPORARY WORKAROUND setup_logger
+
+def check_celltypist_format(indata):
+    if np.abs(np.expm1(indata[0]).sum()-10000) > 1:
+        return False
+    return True
+
+def set_var_index(adata, var_name):
+    adata.var.index = [re.sub("\\.[0-9]+$", "", s) for s in adata.var[var_name]]
+    return adata
+
+def main(par):
+    
+    if bool(par["model"]) == bool(par["reference"]):
+        raise ValueError("Make sure to provide either 'model' or 'reference', but not both.")
```

Comment:
Why not?
```suggestion
    if par["model"] and par["reference"]:
        raise ValueError("Make sure to provide either 'model' or 'reference', but not both.")
```

Reply:
to handle XOR so to raise an error when both are provided as well as none

Reply:
Ah, my bad. I completely misinterpreted what you were trying to do! If you are casting to bool, I would propose to use:

```
not bool(par["model"]) ^ bool(par["reference"])
```
because
```
>>> 1 ^ 0 # One defined
1
>>> 0 ^ 0 # None defined
0
>>> 0 ^ 1 # Other one defined
1
>>> 1 ^ 1 # Both defined
0
>>>
```
so 
```
>>> not 1 ^ 0 # One defined
False
>>> not 0 ^ 0 # None defined
True
>>> not 0 ^ 1 # Other one defined
False
>>> not 1 ^ 1 # Both defined
True
```

`^` is functionally the same as `==` in case there are two values, but if we would like to extend this to three or more values it also works. Also, I think the intention of the `^` operator is more clear, but thats just my opinion

Anyway, I prefer to be explicit here and not cast to bool, because thats just the easiest to read.

```
if (not par["model"] and not par["reference"]) or (par["model"] and par["reference"]):
```